### PR TITLE
Single User Creation Added in Admin.

### PR DESCRIPTION
**Pull request by RishabhJain2018**

### CODE REVIEW

File: profiles/views.py [hunk 1]
```diff
@@ -246,16 +247,76 @@ def post(self, request):
...
-            group = Group.objects.get(name="student")
+            group = str(request.POST.get("group"))
+            groups = Group.objects.get(name=group)
 
             if user_start < user_end:
                 user_no = ["%.3d" % users for users in range(user_start, user_end+1)]
+                # response = HttpResponse(content_type='text/csv')
+                # response['Content-Disposition'] = 'attachment; filename=New_Users.csv'
+                # writer = csv.writer(response)
+                # writer.writerow(['username', 'password', 'course', 'branch', 'year'])
```
Comment (DESHRAJ) [inline profiles/views.py]: Please remove the comments.

File: profiles/views.py [hunk 2]
```diff
@@ -246,16 +247,76 @@ def post(self, request):
...
+    def get(self, request):
+        username = request.user
+
+        if username.is_superuser:
+            return render(request, "singleuserform.html")
+        else:
+            return render(request, "404.html")
+
+    def post(self, request):
+        username = request.user
```
Comment (DESHRAJ) [inline profiles/views.py]: replace `username` with `user`

Comment (RishabhJain2018) [inline profiles/views.py]: yeah sure sir.

File: profiles/views.py (hunk 1 above)
Comment (RishabhJain2018) [inline profiles/views.py]: okk sir.

File: HTTP_200/static/js/main.js [hunk 3]
```diff
@@ -901,3 +901,20 @@ else{
 
 });
 
+
+function ValidateBranch(){
+
+  var course = document.getElementById("course").value
```
Comment (DESHRAJ) [inline HTTP_200/static/js/main.js]: Either use jquery like this `var course = $("#course").val();` or put semicolon at the end.

File: HTTP_200/templates/admin/auth/change_list.html [hunk 4]
```diff
@@ -56,14 +56,20 @@
...
+                <!-- {% blocktrans with cl.opts.verbose_name as name %}Add {{ name }}{% endblocktrans %} -->
+              <!-- </a> -->
+            </li>
+            <!-- <li>
               {% url cl.opts|admin_urlname:'add' as add_url %}
               <a href="{% add_preserved_filters add_url is_popup to_field %}" class="addlink">
                 {% blocktrans with cl.opts.verbose_name as name %}Add {{ name }}{% endblocktrans %}
               </a>
-            </li>
+            </li> -->
```
Comment (DESHRAJ) [inline HTTP_200/templates/admin/auth/change_list.html]: Please remove all comments.

File: HTTP_200/templates/singleuser.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<html>
+<head>
+	<title> BulkUser</title>
+</head>
+<body>
+	<p>All the users have been created.</p>
+	<p><a href="/admin/"> Return to admin pannel again </a></p> 
+	<p><a href="/students/create/single"> Create More Users </a></p>
+	<p><a href="/admin/auth/user/"> See the created users </a></p>
```
Comment (DESHRAJ) [inline HTTP_200/templates/singleuser.html]: Don't use hard coded values for urls.

File: HTTP_200/templates/admin/auth/change_list.html (hunk 4 above)
Comment (RishabhJain2018) [inline HTTP_200/templates/admin/auth/change_list.html]: okk sir.

File: HTTP_200/urls.py [hunk 6]
```diff
@@ -24,6 +24,8 @@
     # url(r'^api/', include('feeds.urls')),
     # url(r'', include('rest_framework.urls', namespace='rest_framework'))
     url(r'^students/create/$', BulkUser.as_view(), name='bulk_students_create'),
+    url(r'^students/create/single/$', SingleUser.as_view(), name='single_user_create'),
```
Comment (DESHRAJ) [inline HTTP_200/urls.py]: line 26 and 27 should be interchanged and line 28 needs to be removed(blank line)

File: HTTP_200/templates/singleuser.html (hunk 5 above)
Comment (RishabhJain2018) [inline HTTP_200/templates/singleuser.html]: okk sir.. I will be updating this page as it is only html.

File: profiles/views.py [hunk 7]
```diff
@@ -20,6 +20,7 @@
 from django.contrib.auth.models import Group
 
 import permissions
+# import csv
```
Comment (DESHRAJ) [inline profiles/views.py]: Please remove comments.

File: profiles/views.py [hunk 8]
```diff
@@ -237,25 +238,69 @@ def get(self, request):
...
+            else:
+                new_user = User.objects.create_user(username=admission_format, password=admission_format)
+                group.user_set.add(new_user)
+                if (group == "student"):
+                    StudentDetail.objects.create(user=new_user, branch=branch, course=course, year=year)
+                else:
+                    FacultyDetail.objects.create(user=new_user)
+            return render(request, "singleuser.html")
+        else:
+            return render(request, "404.html")
```
Comment (DESHRAJ) [inline profiles/views.py]: The backend code looks GREAT. Just little changes you need to do on the client side and its all done. :smiley:

File: HTTP_200/templates/singleuser.html (hunk 5 above)
Comment (RishabhJain2018) [inline HTTP_200/templates/singleuser.html]: @DESHRAJ sir where is this `/admin/auth/user` url located ? I couldn't find it.

**Pull request merged**